src/Nav.js

```diff
@@ -1,359 +1,233 @@
-          'prop to the parent `<TabContainer>`.'
-      );
-
-      id = tabContainer.getTabId(eventKey);
-      controls = tabContainer.getPaneId(eventKey);
-    }
-
-    if (navRole === 'tablist') {
-      role = role || 'tab';
-      onKeyDown = createChainedFunction(
-        event => this.handleTabKeyDown(onSelect, event),
-        onKeyDown
-      );
-      tabIndex = active ? tabIndex : -1;
-    }
+  handleKeyDown = event => {
+    const { onKeyDown, onSelect } = this.props;
+    if (onKeyDown) onKeyDown(event);
 
-    return {
-      id,
-      role,
-      onKeyDown,
-      'aria-controls': controls,
-      tabIndex
-    };
-  }
-
-  handleTabKeyDown(onSelect, event) {
     let nextActiveChild;
-
-    switch (event.keyCode) {
-      case keycode.codes.left:
-      case keycode.codes.up:
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
         nextActiveChild = this.getNextActiveChild(-1);
         break;
-      case keycode.codes.right:
-      case keycode.codes.down:
+      case 'ArrowRight':
+      case 'ArrowDown':
         nextActiveChild = this.getNextActiveChild(1);
         break;
       default:
-        // It was a different key; don't handle this keypress.
         return;
     }
+    if (!nextActiveChild) return;
 
     event.preventDefault();
-
-    if (onSelect && nextActiveChild && nextActiveChild.props.eventKey != null) {
-      onSelect(nextActiveChild.props.eventKey);
-    }
-
+    onSelect(nextActiveChild.dataset.rbEventKey);
     this._needsRefocus = true;
-  }
+  };
 
-  isActive({ props }, activeKey, activeHref) {
-    if (
-      props.active ||
-      (activeKey != null && props.eventKey === activeKey) ||
-      (activeHref && props.href === activeHref)
-    ) {
-      return true;
-    }
-
-    return props.active;
-  }
+  attachRef = ref => {
+    this.listNode = ref;
+  };
 
   render() {
     const {
-      stacked,
-      justified,
+      bsPrefix,
+      variant,
+      fill,
+      justify,
       onSelect,
-      role: propsRole,
-      navbar: propsNavbar,
-      pullRight,
-      pullLeft,
+      navbar,
       className,
       children,
+      componentClass: Component,
       ...props
     } = this.props;
 
-    const tabContainer = this.context.$bs_tabContainer;
-    const role = propsRole || (tabContainer ? 'tablist' : null);
-
-    const { activeKey, activeHref } = this.getActiveProps();
-    delete props.activeKey; // Accessed via this.getActiveProps().
-    delete props.activeHref; // Accessed via this.getActiveProps().
-
-    const [bsProps, elementProps] = splitBsProps(props);
+    delete props.activeKey;
+    delete props.onSelect;
+    delete props.getControlledId;
+    delete props.getControllerId;
 
     const classes = {
-      ...getClassSet(bsProps),
-      [prefix(bsProps, 'stacked')]: stacked,
-      [prefix(bsProps, 'justified')]: justified
+      [bsPrefix]: !navbar,
+      [`navbar-nav`]: navbar,
```

Comment:
ouch... can we read "navbar" from the context provider?

Reply:
yes upcoming in the navbar PR